pkg/vegeta/format.go

```diff
@@ -0,0 +1,126 @@
+package vegeta
+
+// MetaInfo provide map to store meta information for Format
+type MetaInfo map[string]string
+
+// Format defines a type for the format query param
+type Format interface {
+	// fmt.Stringer
+	String() string
+
+	Meta() MetaInfo
+	SetMeta(key, value string)
+}
+
+// NewFormat returns a new Format in accordance with the provided type string, by default it is JSON
+func NewFormat(typ string) Format {
+	switch typ {
+	case "json":
+		return NewJSONFormat()
+	case "text":
+		return NewTextFormat()
+	case "binary":
+		return NewBinaryFormat()
+	case "histogram":
+		return NewHistogramFormat()
+	}
+	return NewJSONFormat() // default
+}
+
+// JSONFormat typedef for query param "json"
+type JSONFormat string
+
+// NewJSONFormat returns a new Format of JSON type
+func NewJSONFormat() *JSONFormat {
+	f := JSONFormat("json")
+	return &f
+}
+
+// SetMeta will set the meta information
+func (j *JSONFormat) SetMeta(key, value string) {
```

Comment:
Add a comment stating this is a NO-OP for `JSONFormat`, `TextFormat` and `BinaryFormat`, so that the next contributor can understand why the blocks are left unimplemented.
Same for `Meta()` for these three.